src/shared/contexts/DatabaseInfoMaps.tsx

```diff
@@ -0,0 +1,69 @@
+import { createContext, ReactNode, useEffect, useMemo } from 'react';
+
+import {
+  DatabaseInfoMaps,
+  getDatabaseInfoMaps,
+} from '../../uniprotkb/utils/database';
+import apiUrls from '../config/apiUrls';
+import useDataApi from '../hooks/useDataApi';
+import * as logging from '../utils/logging';
+
+import { UniProtKBColumn } from '../../uniprotkb/types/columnTypes';
+import { DatabaseInfo } from '../../uniprotkb/types/databaseRefs';
+
+export const DatabaseInfoMapsContext = createContext<DatabaseInfoMaps | null>(
+  null
+);
+
+type DatabaseInfoMapsProviderProps = {
+  children: ReactNode;
+};
+
+export const DatabaseInfoMapsProvider = ({
+  children,
+}: DatabaseInfoMapsProviderProps) => {
+  const { data } = useDataApi<DatabaseInfo>(apiUrls.allUniProtKBDatabases);
+  // Check against columnTypes enum
+  // databaseInfo - columnTypes
+  // columnTypes - databaseInfo
+  const databaseInfoMaps = useMemo(
+    () => (data ? getDatabaseInfoMaps(data) : data),
+    [data]
+  );
+
+  // Sanity check for dynamic database info and static column definition
+  useEffect(() => {
+    if (data) {
+      const definedColumns = new Set<string>();
+      Object.values(UniProtKBColumn).forEach((column) => {
```

Comment:
What do you think about extracting the sanity check into its own function and then also testing this?

Reply:
Yep, will get on it.